fs/accounting.go

```diff
@@ -394,12 +444,12 @@ func (acc *Account) read(in io.Reader, p []byte) (n int, err error) {
 	// Get the token bucket in use
 	tokenBucketMu.Lock()
 	tb := tokenBucket
-	tokenBucketMu.Unlock()
 
 	// Limit the transfer speed if required
 	if tb != nil {
 		tb.Wait(int64(n))
 	}
+	tokenBucketMu.Unlock()
```

Comment:
It would be nice to know the full story here, but I'll leave it how it is and take your word for it :-)